src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -198,4 +198,53 @@ theorem ule_eq_not_ult (x y : BitVec w) : x.ule y = !y.ult x := by
 theorem ule_eq_carry (x y : BitVec w) : x.ule y = carry w y (~~~x) true := by
   simp [ule_eq_not_ult, ult_eq_not_carry]
 
+/-- If two bitvectors have the same `msb`, then signed and unsigned comparisons coincide -/
+theorem slt_eq_ult_of_msb_eq {x y : BitVec w} (h : x.msb = y.msb) :
+    x.slt y = x.ult y := by
+  simp only [BitVec.slt, toInt_eq_msb_cond, BitVec.ult, decide_eq_decide, h]
+  cases y.msb <;> simp only [↓reduceIte, Int.ofNat_lt, Int.sub_lt_sub_right_iff]
```

Comment:
I think we can just make `Int.sub_lt_sub_left_iff` (and `right`) `@[simp]` lemmas, and then just 
```suggestion
  simp only [BitVec.slt, toInt_eq_msb_cond, BitVec.ult, h]
  cases y.msb <;> simp
```